control-plane/api-gateway/consul/config_entry_translation.go

```diff
@@ -0,0 +1,91 @@
+package consul
+
+import (
+	"os"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/hashicorp/consul-k8s/control-plane/namespaces"
+	"github.com/hashicorp/consul/api"
+	capi "github.com/hashicorp/consul/api"
+)
+
+const (
+	metaKeyManagedBy       = "managed-by"
+	metaValueManagedBy     = "consul-k8s-gateway-controller"
+	metaKeyKubeNS          = "k8s-namespace"
+	metaKeyKubeServiceName = "k8s-service-name"
```

Comment:
nit: I believe most of these are already defined elsewhere, so I'm pretty sure we could re-use them.

Reply:
I'm pretty sure that every place we use those they have the exact same value (`metaKeyKubeServiceName` appears to be defined in multiple packages)... so I'd definitely lean towards consolidation but I'd also be fine doing that later too.

Reply:
I'm down to do that in a separate PR because that'll be a change that touches a lot of files